src/classes/base-page.ts

```diff
@@ -0,0 +1,34 @@
+import { Platform, NavController } from 'ionic-angular';
+import { AuthenticationProvider } from '../providers/authentication/authentication';
+
+export abstract class BasePageComponent {
```

Comment:
do we know what this keyword does?

Reply:
abstract ? - Basically we are unable to create an instance of it, the only way to use it is to create a sub class (All pages will extend this class). We are basically using it give the same functionality to multiple pages and if needed in the future can use it to force pages to implement certain methods.